src/kani-compiler/rustc_codegen_kani/src/codegen/function.rs

```diff
@@ -263,20 +313,64 @@ impl<'tcx> GotocCtx<'tcx> {
             mangled_name,
             original_file: loc.filename().unwrap(),
             original_line: loc.line().unwrap().to_string(),
+            unwind_value: None,
         };
 
-        self.proof_harnesses.push(harness);
+        harness
+    }
+
+    /// Unwind strings of the format 'unwind_x' and the mangled names are to be parsed for the value 'x'
+    fn handle_kanitool_unwind(&mut self, attr: &Attribute, harness: &mut HarnessMetadata) {
```

Comment:
Since we are planning to eventually support unwind outside of the metadata, I would make this function independent from harness metadata. I.e.:

```rust
fn handle_unwind(&mut self, attrs: &BTreeMap<&str, &Attribute>) -> Option<UnwindMetadata> {
}
```

BTW, see my other comment about `UnwindMetadata` type.

Reply:
Not sure I understand this comment, if we're putting unwind attributes elsewhere from proof harnesses (imo, still an 'if'), that's going to require a ton of design work, the choices of which we can't really anticipate. This approach seems simple. Trying to anticipate what we might do later is overly complex for the time being.

Reply:
Oh now I see what you mean. I'm not sure if it is cleaner though. The need to give good error messages demands we spot duplicates and things like that. If we try to make this function "cleaner" then we end up pushing a lot of unwind-specific logic (like detecting duplicate unwind attributes) up to the calling function instead of here in the 'unwind' handler.

This organization lets that calling function have that nice, clean "loop over kanitool attributes, call handler function" and that's it.

Reply:
Anyway, not a biggie. Let's move on.